phpBB/phpbb/messenger/method/phpbb_email.php

```diff
@@ -0,0 +1,637 @@
+	}
+
+	/**
+	 * Set up subject for mail
+	 *
+	 * @param string	$subject	Email subject
+	 * @return void
+	 */
+	public function subject($subject = '')
+	{
+		parent::subject(trim($subject));
+		$this->email->subject($this->subject);
+	}
+
+	/**
+	 * Set up extra mail headers
+	 *
+	 * @param string	$header_name	Email header name
+	 * @param string	$header_value	Email header body
+	 * @return void
+	 */
+	public function header($header_name, $header_value)
+	{
+		$header_name = $header_name;
+		$header_value = $header_value;
+
+		// addMailboxListHeader() requires value to be array
+		if ($this->get_header_method($header_name) == 'addMailboxListHeader')
+		{
+			$header_value = [$header_value];
+		}
+		$this->headers->addHeader($header_name, $header_value);
+	}
+
+	/**
+	 * Adds X-AntiAbuse headers
+	 *
+	 * @param \phpbb\config\config	$config		Config object
+	 * @param \phpbb\user			$user		User object
+	 * @return void
+	 */
+	public function anti_abuse_headers($config, $user)
+	{
+		$this->header('X-AntiAbuse', 'Board servername - ' . $config['server_name']);
+		$this->header('X-AntiAbuse', 'User_id - ' . $user->data['user_id']);
+		$this->header('X-AntiAbuse', 'Username - ' . $user->data['username']);
+		$this->header('X-AntiAbuse', 'User IP - ' . $user->ip);
+	}
+
+	/**
+	 * Set the email priority
+	 *
+	 * Possible values are:
+	 * Email::PRIORITY_HIGHEST = 1
+	 * Email::PRIORITY_HIGH = 2
+	 * Email::PRIORITY_NORMAL = 3
+	 * Email::PRIORITY_LOW = 4
+	 * Email::PRIORITY_LOWEST = 5
+	 *
+	 * @param int	$priority	Email priority level
+	 * @return void
+	 */
+	public function set_mail_priority($priority = Email::PRIORITY_NORMAL)
+	{
+		$this->email->priority($priority);
+	}
+
+	/**
+	 * Detect proper Header class method to add header
+	 *
+	 * @param string	$name	Email header name
+	 * @return string
+	 */
+	protected function get_header_method(string $name)
+	{
+		$parts = explode('\\', self::HEADER_CLASS_MAP[strtolower($name)] ?? UnstructuredHeader::class);
+		$method = 'add'.ucfirst(array_pop($parts));
+		if ('addUnstructuredHeader' === $method)
+		{
+			$method = 'addTextHeader';
+		}
+		else if ('addIdentificationHeader' === $method)
+		{
+			$method = 'addIdHeader';
+		}
+
+		return $method;
+	}
+
+	/**
+	 * Set email headers
+	 *
+	 * @return bool
+	 */
+	protected function build_header()
+	{
+		$headers = [];
+
+		$board_contact = $this->config['board_contact'];
+		if (empty($this->email->getReplyTo()))
+		{
+			$this->replyto($board_contact);
+			$headers['Reply-To'] =  $this->replyto;
+		}
+
+		if (empty($this->email->getFrom()))
+		{
+			$this->from($board_contact);
+			$headers['From'] = $this->from;
+		}
+
+		$headers['Return-Path'] = new Address($this->config['board_email']);
+		$headers['Sender'] = new Address($this->config['board_email']);
+		$headers['X-Priority'] = sprintf('%d (%s)', $this->mail_priority, self::PRIORITY_MAP[$this->mail_priority]);
```

Comment:
Can't believe this would will be needed with Symfony mailer and I don#t think it is?

Reply:
Headers aren't set at this point yet. Using Symfony `Email()->priority()` here would mess things as other headers got set later.